### PR TITLE
GCP - CIAC - 5981

### PR DESCRIPTION
**Pull request by johnnywilkes**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://jira-hq.paloaltonetworks.local/browse/CIAC-5981

## Description
A customer would like to be able to use a GCP service account scoped at the organization level with the Xpanse "Cortex ASM - ASM Alert" playbook.  Please see my video for more information.

## Screenshots
See video

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [X] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [ ] Tests
- [X] Documentation

### CODE REVIEW

**Comment by johnnywilkes** [comment]
@daryakoval , i shared a demo video with you via google drive

File: Packs/GoogleCloudCompute/Integrations/GoogleCloudCompute/GoogleCloudCompute.py [hunk 1]
```diff
@@ -4370,19 +4391,44 @@ def aggregated_list_instances_ip(args: Dict[str, Any]) -> CommandResults:
...
+    default_search_scope = demisto.params().get('default_search_scope')
+    # 'default_search_scope' param was set use it for scope, else use the project in the service account.
+    # 'compute.googleapis.com/Instance' asset-type needed to find static and ephemeral public IPs.
+    if default_search_scope:
+        request_asset = get_asset().v1().searchAllResources(
+            scope=default_search_scope,
+            assetTypes='compute.googleapis.com/Instance',
+            query=f"additionalAttributes.externalIPs={ip}"
+        )
+
```
Comment (BigEasyJ) [inline Packs/GoogleCloudCompute/Integrations/GoogleCloudCompute/GoogleCloudCompute.py:4404]:
```suggestion
```

File: Packs/GoogleCloudCompute/Integrations/GoogleCloudCompute/GoogleCloudCompute.py [hunk 2]
```diff
@@ -4370,19 +4391,44 @@ def aggregated_list_instances_ip(args: Dict[str, Any]) -> CommandResults:
...
+    else:
+        request_asset = get_asset().v1().searchAllResources(
+            scope=f"projects/{SERVICE_ACT_PROJECT_ID}",
+            assetTypes='compute.googleapis.com/Instance',
+            query=f"additionalAttributes.externalIPs={ip}"
+        )
+    response_asset = request_asset.execute()
+    if response_asset:
```
Comment (BigEasyJ) [inline Packs/GoogleCloudCompute/Integrations/GoogleCloudCompute/GoogleCloudCompute.py:4412]:
```suggestion
    
    if response_asset:
```

File: Packs/GoogleCloudCompute/Integrations/GoogleCloudCompute/GoogleCloudCompute.py [hunk 3]
```diff
@@ -4370,19 +4391,44 @@ def aggregated_list_instances_ip(args: Dict[str, Any]) -> CommandResults:
         CommandResults: A ``CommandResults`` object that is then passed to ``return_results``, that contains instance
         details.
     """
-    project = SERVICE_ACT_PROJECT_ID
     ip = args.get('ip')
+    default_search_scope = demisto.params().get('default_search_scope')
```
Comment (BigEasyJ) [inline Packs/GoogleCloudCompute/Integrations/GoogleCloudCompute/GoogleCloudCompute.py:4395]: Do you feel like this needs any error handling here or in the `if/else` for what the potential input could be?

**Review by BigEasyJ — CHANGES_REQUESTED** [review]

File: Packs/GoogleCloudCompute/Integrations/GoogleCloudCompute/GoogleCloudCompute.py (hunk 1 above)
Comment (johnnywilkes) [reply Packs/GoogleCloudCompute/Integrations/GoogleCloudCompute/GoogleCloudCompute.py:4404]: this is just remove a blank line?

**Review by daryakoval — APPROVED** [review]

**Pull request merged**